R_scripts/slope_calc_master_script.R

```diff
@@ -0,0 +1,299 @@
+      metadata(cov_window)$sample_filename <- metadata(coverage_data_rev[[sample_index]])$sample_filename
+      metadata(cov_window)$chromosome <- metadata(coverage_data_fwd[[sample_index]])$chromosome
+      return(cov_window)
+    }, introns_per_strand_and_chromosome_rev$start_position_measured,
+      introns_per_strand_and_chromosome_rev$end_position_measured,
+      mc.cores = nr_of_cores)
+  })
+  
+  # Combine forward and reverse strand data
+  if (length(intron_coverage_data_fwd[[1]]) == 0) intron_coverage_data_fwd <- NULL else intron_coverage_data_fwd <- lapply(1:size_of_annot_fwd, function(y) lapply(intron_coverage_data_fwd, function(x) x[[y]]))
+  if (length(intron_coverage_data_rev[[1]]) == 0) intron_coverage_data_rev <- NULL else intron_coverage_data_rev <- lapply(1:size_of_annot_rev, function(y) lapply(intron_coverage_data_rev, function(x) x[[y]]))
+  
+  chromosome_intron_coverage_data <- c(intron_coverage_data_fwd, intron_coverage_data_rev)  # Combine forward and reverse strand data
+  return(chromosome_intron_coverage_data)  # Return combined data
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #  # # # # # # # # # # # # # # # # # # # # # # # # # # 
+# Function to fit linear models to intron coverage data for a specific chromosome                                      #
+# Args:                                                                                                                #
+#   chromosome: Chromosome name to filter coverage data.                                                               #
+#   intron_coverage_data: List of intron coverage data for different chromosomes.                                      #
+#   intron_info: Data frame containing intron information.                                                             #
+#   nr_of_cores: Number of cores to use for parallel processing (default is 6).                                        #
+#   component_from_lm: Component of the linear model to return (default is "all").                                     #
+# Returns:                                                                                                             #
+#   Data frame with intron information combined with linear model results (y-intercept, x-intercept, slope, p-value).  #
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #  # # # # # # # # # # # # # # # # # # # # # # # # # #
+lm_fit_intron_coverage_per_chromosome <- function(chromosome, 
+                                                  intron_coverage_data, 
+                                                  intron_info,
+                                                  nr_of_cores = 6, 
+                                                  component_from_lm = "all") 
+  {
+  # Subselect introns for chromosome and strand
+  introns_per_strand_and_chromosome_fwd <- filter_introns_by_chromosome_and_strand(intron_info_file = intron_info, chromosome = chromosome, strand_sense = "plus")
```

Comment:
I would name the strands as '+' and '-' to be consistent with .bed format notation.